mqtt_as/mqtt_as.py

```diff
@@ -329,12 +328,28 @@ def close(self):
         if self._sock is not None:
             self._sock.close()
 
+    async def _await_pid(self, pid):
+        t = ticks_ms()
+        self.dprint("awaiting pid", pid)
+        while pid in self.rcv_pids:  # local copy
+            if self._timeout(t) or not self.isconnected():
+                self.dprint("timeout pid", pid, "isconnected", self.isconnected())
+                break  # Must repub or bail out
+            await asyncio.sleep_ms(100)  # putting sleep at the end will solve the issue
+            # of not recognizing received pids in the event of disconnect
+        else:
+            self.dprint("pid", pid, "not in rcv_pids anymore")
+            return True  # PID received. All done.
+        return False
+
```

Comment:
Common approach for awaiting a SUB/PUBACK pid for publish and subscribe routines. Reduces code duplication as both operations can use the same mechanism.